### PR TITLE
[SDPAP-7564] Adds type icons to the media view

### PR DESCRIPTION
**Pull request by vincent-gao**
### Jira
https://digital-vic.atlassian.net/browse/SDPAP-7564

### Change
1. Upgrade the icons to remain consistent with the Ripple design.
2. Applied icons to the `media` view.

image

### CODE REVIEW

File: tide_media.module [hunk 1]
```diff
@@ -321,9 +324,24 @@ function tide_media_views_pre_render(ViewExecutable $view) {
           if (strpos($file_mime, 'presentation')) {
             $value->_relationship_entities['field_media_file_target_id']->set('filemime', 'presentation');
           }
-          if (strpos($file_mime, 'text/plain')) {
+          if ($file_mime == 'text/plain') {
             $value->_relationship_entities['field_media_file_target_id']->set('filemime', 'plaintext');
           }
+          if ($file_mime == 'application/vnd.ms-excel.sheet.macroEnabled.12') {
```
Comment (krakerag) [inline tide_media.module:330]: Are you sure this is correct MIME for Excel? It seems awfully specific

File: tide_media.module [hunk 2]
```diff
@@ -312,6 +312,9 @@ function tide_media_views_pre_render(ViewExecutable $view) {
           if (strpos($file_mime, 'msword') || strpos($file_mime, 'wordprocessingml.document')) {
             $value->_relationship_entities['field_media_file_target_id']->set('filemime', 'msword');
           }
+          if (strpos($file_mime, 'wordprocessingml.document')) {
```
Comment (krakerag) [inline tide_media.module:315]: Same here with word... hmm.

**Review by krakerag — COMMENTED** [review]
Looks good @vincent-gao I only have queries about the mime type detection of icons

A few web resources say that there can be multiple mime types for word and excel style docs - see here for example

https://stackoverflow.com/questions/4212861/what-is-a-correct-mime-type-for-docx-pptx-etc

The way we catch them in the pre-render hooks means some will be missing, so they may not get the right icon.

File: tide_media.module (hunk 1 above)
Comment (vincent-gao) [reply tide_media.module:330]: hey @krakerag ,thanks for the CR
yes, it is not looking good! 
The mimetype for `.xlsb` file format is `application/vnd.ms-excel.sheet.macroEnabled.12`. Despite the awful name, we require it to obtain precise icons.

Comment (krakerag) [reply tide_media.module:330]: OK - just wanted to double check. Carry on!

**Review by krakerag — APPROVED** [review]

**Pull request merged**